src/bufr/BufrParser/Query/Tokenizer.h

```diff
@@ -0,0 +1,386 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <regex>  // NOLINT
+#include <typeinfo>
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <set>
+
+#include "eckit/exception/Exceptions.h"
+
+
+namespace Ingester {
+namespace bufr {
+
+    class Token;
+    class QueryToken;
+    typedef std::vector<std::shared_ptr<Token>> Tokens;
+
+    /// \brief Token abstract base class
+    class Token {
+     public:
+        Token() = default;
+        virtual ~Token() = default;
+
+        /// \brief Get the string representation of the token
+        std::string str() const  { return str_; }
+
+        /// \brief Get the debug string for the token.
+        virtual std::string debugStr() const = 0;
+
+        /// \brief Get the subtokens for this token.
+        virtual std::vector<std::shared_ptr<QueryToken>> queryTokens() const
```

Comment:
So would a subtoken be when we're looking at something like `*/BRIT{2}/TMBR` where TMBR  is the subtoken?

Reply:
So this returns a list of QueryToken objects. So for example if you've got a MultiQuery like `[*/BRITCSTC/TMBR, */BRIT/TMBR]` then the query tokens are `*/BRITCSTC/TMBR` and `*/BRIT/TMBR`.